server/main.ts

```diff
@@ -69,6 +70,26 @@ loadConfig((loadedInitialConfig) => {
     )
   );
 
+  const wss = new WebSocket.Server({ noServer: true });
+  instance.on('upgrade', (req, socket, head) => {
+    const { upgrade, connection } = req.headers;
+    logger.trace(
+      { headers: { upgrade, connection } },
+      `Upgrade request received for ${req.url}`
+    );
+
+    if (upgrade === 'websocket') {
+      wss.handleUpgrade(req, socket, head, (ws) => {
+        const res = new http.ServerResponse(req) as strimziUIResponseType;
+        // add/mark the request as a websocket request
+        req.isWs = true;
+        res.ws = ws;
+        // call the express app as usual
+        expressApp(req, res);
```

Comment:
Please see previous comment re breakout - not sure how safe this is for GC purposes that we lose reference to the ws (to be able to close it etc)

Reply:
@ajborley / @nictownsend has this chat happened yet? Would be keen to participate. If the `ws` was placed in `res.locals`, it will be scoped to the life of the request (ie socket lifetime). Express will then discard the `locals`, `ws` included. Thoughts?

Reply:
I think this will be superseded when Matt's websocket handling changes go in, so I've left it as-is for now.

Reply:
Yep - I will update the approach in my next PR